src/components/BookCard.jsx

```diff
@@ -135,6 +199,16 @@ function BookCard({ user_uid, image, genres, title, price, rating, id }) {
             <></>
           )}
         </div>
+        <div className="flex justify-start items-center">
+          <ReactStars
+            className=""
+            size={20}
+            isHalf={true}
+            value={currentRating}
+            onChange={(newR) => setRating(newR)}
+          />
```

Comment:
The user shouldn't be able to rate if he's not signed in, but he could see the rates only

Reply:
It is already implemented